crates/gitbutler-stack/src/stack_branch.rs

```diff
@@ -140,36 +138,18 @@ impl StackBranch {
             let head_commit = repository
                 .find_reference(&self.remote_reference(&remote_name)?)?
                 .peel_to_commit()?;
-            let target_commit = repository
-                .find_reference(default_target.branch.to_string().as_str())?
-                .peel_to_commit()?;
-            let merge_base = repository.merge_base(head_commit.id(), target_commit.id())?;
             repository
-                .log(head_commit.id(), LogUntil::Commit(merge_base), false)?
+                .log(head_commit.id(), LogUntil::Commit(previous_head), false)?
```

Comment:
Logging to the merge base includes extra commits from previous branches which we don't want.